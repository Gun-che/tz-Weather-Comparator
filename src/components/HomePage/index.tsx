import React from 'react'
import s from './index.module.scss'

export const Home: React.FC = () => {
  return (
    <section className={s.wrap}>
      Home
    </section>
  )
}

export default Home