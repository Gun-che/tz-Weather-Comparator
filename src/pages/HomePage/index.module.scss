.wrap {
  text-align: left;
  line-height: 250%;
  h2 {
    font-size: 22px;
  }
  h3 {
    font-size: 20px;
  }
  p,
  li,
  p {
    font-size: 18px;
  }
  li {
    i {
      font-size: 15px;
      margin-right: 7px;
    }
  }
}
