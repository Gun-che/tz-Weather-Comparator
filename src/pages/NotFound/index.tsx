import React from 'react'
import s from './index.module.scss'

export const NotFound: React.FC = () => {
  return (
    <>
      <div className={s.wrap + ' containerCust'}>
        404
        <h3>not found</h3>
      </div>
    </>
  )
}


export default NotFound