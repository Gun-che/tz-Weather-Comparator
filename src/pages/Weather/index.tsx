import React, { SyntheticEvent } from 'react'
import { Route, Link, useLocation } from 'react-router-dom'
import { PropsFromRedux } from '../../containers/WeatherContainer'
import { CurrentWeather } from './components/CurrentWeather'
import { errorMsg } from '../../helpers/messages'
import { HourlyForecast } from './components/HourlyForecast'
import { LoadingThin } from '../../components/Loading'
import { DailyForecast } from './components/DailyForecast'

import s from './index.module.scss'
import { ErrorMessage } from '../../components/ErrorMessage'

export const Weather: React.FC<PropsFromRedux> = ({
  getWeather,
  daily,
  hourly,
  data,
  message,
  code,
  isFetching
}) => {

  const tmpLocation = useLocation()
  console.log(tmpLocation)

  const [searchValue, setSearchValue] = React.useState<string>('')
  const [formMessage, setFormMessage] = React.useState<string>('')
  const [tabsClass, setTabsClass] = React.useState(s.cw)

  React.useEffect(() => {
    const loc = tmpLocation.pathname;
    // eslint-disable-next-line @typescript-eslint/no-unused-expressions
    loc === '/weather' && tabsClass !== s.cw ? setTabsClass(s.cw) :
      loc === '/weather/hourly' ? setTabsClass(s.hf) :
        loc === '/weather/daily' ? setTabsClass(s.df) :
          null

  }, [tabsClass, tmpLocation.pathname])

  const onSubmit = (e: SyntheticEvent<HTMLFormElement>) => {
    e.preventDefault();

    if (searchValue) {
      getWeather({ city: searchValue })
      formMessage && setFormMessage('')

    } else {
      setFormMessage(errorMsg.empty['ru'])
    }
  }

  const onChange = (e: SyntheticEvent<HTMLInputElement>) => {
    setSearchValue(e.currentTarget.value)
  }

  const tmpMsg = (): string => {
    const fullMsg = errorMsg[code] ? errorMsg[code]['ru'] : message
    return fullMsg || formMessage ? `${fullMsg} ${formMessage}` : ''
  }

  return (
    <section className={s.wrap + ' container'}>
      <ErrorMessage message={tmpMsg()} />
      <div className={tabsClass + ' card-tabs ' + s.tabs}>
        <ul className="tabs">
          <li className={s.cw + " tab"}>
            <Link to="/weather">Текущая погода</Link>
          </li>
          <li className={s.hf + " tab"}>
            <Link to="/weather/hourly">Почасовой прогноз</Link>
          </li>
          <li className={s.df + " tab"}>
            <Link to="/weather/daily">Прогноз на неделю</Link>
          </li>
        </ul>
      </div>
      <form className={s.form} onSubmit={onSubmit}>
        <div className="input-field">
          <input
            name="search"
            type="text"
            id="search"
            value={searchValue}
            onChange={onChange}
            className="validate"
          />
          <label htmlFor="search">Введите город</label></div>
        <button
          type="submit"
          className="waves-effect waves-light btn-floating blue lighten-3">
          <i className="material-icons left">search</i></button>
      </form>

      <Route path="/weather" exact>
        {data.length ?
          <CurrentWeather data={data[0]} /> :
          isFetching ? <LoadingThin /> :
            <h2 className={s.input}>Введите название города</h2>}
      </Route>

      <Route path="/weather/hourly">
        {hourly.length ?
          <HourlyForecast data={hourly} name={data[0].name} /> :
          isFetching ? <LoadingThin /> :
            <h2 className={s.input}>Введите название города</h2>}
      </Route>

      <Route path="/weather/daily">
        {daily.length ?
          <DailyForecast name={data[0].name} data={daily} /> :
          isFetching ? <LoadingThin /> :
            <h2 className={s.input}>Введите название города</h2>}
      </Route>
    </section>
  )
}