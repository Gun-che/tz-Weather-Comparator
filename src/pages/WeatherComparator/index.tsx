import React, { SyntheticEvent, ChangeEvent } from 'react'

import { PropsFromRedux } from '../../containers/WeatherComparatorContainer'
import { errorMsg } from '../../helpers/messages'
import { ComparisonCard } from '../../components/ComparisonCard'
import { WeatherCard } from '../../components/WeatherCard'
import { CityInput } from '../../components/CityInput'
import { ComparisonInfo } from '../../components/ComparisonInfo'
import s from './index.module.scss'
import { ErrorMessage } from '../../components/ErrorMessage'

export const WeatherComparator: React.FC<PropsFromRedux> = ({
  data,
  getWeather,
  message,
  code,
}) => {

  const [cities, setSities] = React.useState<[string, number][]>([['', 0], ['', 1]])
  const [formMessage, setFormMessage] = React.useState<string>('')
  const [key, setKey] = React.useState<number>(2)

  const onSubmit = (e: SyntheticEvent<HTMLFormElement>) => {
    e.preventDefault();

    if (cities.some(i => i[0])) {
      const post = cities
        .filter(i => i[0])
        .map(i => i[0])

      getWeather(post)
      formMessage && setFormMessage('')

    } else {
      setFormMessage(errorMsg.empty2['ru'])
    }
  }

  const onChange = (e: ChangeEvent<HTMLInputElement & { dataset: { index: number } }>) => {
    const target = e.currentTarget;
    let arr = [...cities];
    arr[target.dataset.index][0] = target.value;

    setSities(arr);
  }

  const onAdd = (e: SyntheticEvent) => {
    setSities([...cities, ['', key]])
    setKey(key + 1)
  }

  const onClose = (i: number): (e: SyntheticEvent) => void => e => {
    let arr = [...cities];
    arr.splice(i, 1);
    setSities(arr)
  }


  const tmpGridClass = (): string => {
    return (data.length === 2 || data.length === 4) ? s.two :
      data.length === 0 ? s.empty : ''
  }

  const tmpMsg = (): string => {
    const fullMsg = errorMsg[code] ? errorMsg[code]['ru'] : message
    return fullMsg || formMessage ?
      `${fullMsg} ${formMessage}` :
      ''
  }

  return (
    <div className={s.wrap + ' ' + tmpGridClass() + ' containerCust'}>

      <div className={s.message}>
        <h2>Введите названия городов, погоду в которых вы хотите сравнить</h2>
        <ErrorMessage message={tmpMsg()} />
      </div>

      <form onSubmit={onSubmit} className={s.form}>
        {cities.map((item, index) => {
          return <div key={item[1]}>
            <CityInput
              onClose={onClose}
              item={item}
              index={index}
              onChange={onChange}
            />
          </div>
        })}
        <div className={s.btns}>
          <button
            onClick={onAdd}
            className="waves-effect waves-light btn blue lighten-3">
            <i className="material-icons left">add_location</i>Add location</button>
          <button
            type="submit"
            className="waves-effect waves-light btn blue lighten-3">
            <i className="material-icons left">search</i>Search</button>
        </div>
      </form>

      {data.length ?
        <><ComparisonCard data={data} />
          <div className={s.infoBlock}><ComparisonInfo data={data} /></div>
        </> : null}

      <div className={s.grid}>
        {data.map((item, index) => {
          return (
            <div key={index}>
              <WeatherCard data={item} />
            </div>
          )
        })}
      </div>
    </div>
  )
}

