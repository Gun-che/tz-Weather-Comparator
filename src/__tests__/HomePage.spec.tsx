import React from 'react'
import { shallow } from 'enzyme'
import { Home } from '../pages/HomePage'

describe('home component', () => {
  describe('home component render', () => {
    const home = shallow(<Home />)

    it('renders properly', () => {
      expect(home).toMatchSnapshot()
    })

    it('render <h2>', () => {
      expect(home.find('h2')).toHaveLength(1)
    })

    it('renders inner <p>', () => {
      expect(home
        .find('h2')
        .first()
        .text()).toEqual('Weather Comparator')
    })

    it('render <h3>', () => {
      expect(home.find('h3')).toHaveLength(2)
    })

    it('renders inner <h2>', () => {
      expect(home
        .find('h3')
        .first()
        .text()).toEqual('Cтек технологий, используемых для разработки:')
    })
  })
})