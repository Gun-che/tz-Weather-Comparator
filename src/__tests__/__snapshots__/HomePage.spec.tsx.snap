// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home component home component render renders properly 1`] = `ShallowWrapper {}`;
